src/classes/ALLO_Allocations_TEST.cls

```diff
@@ -117,6 +117,77 @@ private with sharing class ALLO_Allocations_TEST {
         system.assertEquals(8, queryOpps[0].Amount, 'The opportunity amount should remain unchanged after the exception is caught.');
     }
 
+    /*******************************************************************************************************
+    * @description When Payments are created after the Opportunity and Opportunity Allocations exist,
+    * ensure that the payment allocation percentages match those from the Opportunity Allocations, and
+    * are not overwritten by additional logic in the trigger.
+    * Also check that allocation rounding works correctly and no amounts are allocated to the Default GAU
+    ********************************************************************************************************/
+    static testMethod void pmtAllocationPercentCheck() {
```

Comment:
I also have a larger question about what this test is actually testing. The method that was changed in the `ALLO_Allocations_TDTM` class is `copyRecurringDonationCampaignAllocations()`. Unless I'm mis-reading the TDTM class, I'm fairly sure that method is only called copying default Allocations from a related RD or Campaign to an Opportunity and/or for adjusting the default allocation amounts. This test method doesn't use a Campaign or RD for default Allocations though. Am I misunderstanding how the logic in the trigger is actually working? And if, so does the description of that method name (and maybe the method name) need to be updated to reflect it's true behavior?

Reply:
It appears that the `copyRecurringDonationCampaignAllocations()` method has been repurposed to also copy Payment Allocations from Opportunities, without being renamed. The description includes this detail, but I can update the method name.

Reply:
We probably don't want to take this on for this work item, but I'd say that it's worth having a conversation at some point about completely rebuilding that method so that it delegates individual responsibilities to much smaller methods that do _one_ thing. That would also significantly help with debugging, future updates, and writing unit tests. fyi @jbrandolinisf